101-azure-bastion/azuredeploy.json

```diff
@@ -0,0 +1,242 @@
+                            "sourceAddressPrefix": "*",
+                            "destinationPortRange": "443",
+                            "destinationAddressPrefix": "GatewayManager",
+                            "access": "Allow",
+                            "priority": 120,
+                            "direction": "Outbound"
+                        }
+                    },
+                    {
+                        "name": "bastion-azure-out-allow",
+                        "properties": {
+                            "protocol": "Tcp",
+                            "sourcePortRange": "*",
+                            "sourceAddressPrefix": "*",
+                            "destinationPortRange": "443",
+                            "destinationAddressPrefix": "AzureCloud",
+                            "access": "Allow",
+                            "priority": 130,
+                            "direction": "Outbound"
+                        }
+                    },
+                    {
+                        "name": "bastion-out-deny",
+                        "properties": {
+                            "protocol": "*",
+                            "sourcePortRange": "*",
+                            "sourceAddressPrefix": "*",
+                            "destinationPortRange": "*",
+                            "destinationAddressPrefix": "*",
+                            "access": "Deny",
+                            "priority": 900,
+                            "direction": "Outbound"
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "apiVersion": "2019-02-01",
+            "name": "[parameters('vnet-name')]",
+            "type": "Microsoft.Network/virtualNetworks",
+            "location": "[parameters('location')]",
+            "condition": "[equals(parameters('vnet-new-or-existing'), 'new')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/networkSecurityGroups', variables('nsg-name'))]"
+            ],
+            "properties": {
+                "addressSpace": {
+                    "addressPrefixes": [
+                        "[parameters('vnet-ip-prefix')]"
+                    ]
+                },
+                "subnets": [
+                    {
+                        "name": "[variables('bastion-subnet-name')]",
+                        "properties": {
+                            "addressPrefix": "[parameters('bastion-subnet-ip-prefix')]",
+                            "networkSecurityGroup": {
+                                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('nsg-name'))]"
+                            }
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "apiVersion": "2019-02-01",
+            "type": "Microsoft.Network/virtualNetworks/subnets",
```

Comment:
does it make sense to have this sample always deploy a new vnet?  This pattern is going to be pretty fragile...  For starters, it will be common to have the vnet in another resourceGroup.  As well, the subnet prefix will have to be precise (unused, no overlap).  Users will be more successful in the new pattern if that is appropriate for this scenario.

Reply:
We have a large # of customers who have a need for both scenarios - while it's common to have networking resources, such as VNETs in a different resource group from other resources deployed for applications, AzureBastion and the other deployed resources are all Networking elements, so customers would like deploy in the same resource group.  I'm defaulting to "new" to help ensure success for default deployments and testing, but I think there's value in providing the option for "existing" in this case.

Reply:
Ok, I'll merge this as you have it now - the scenario that will be a challenge here is deploying to a vnet in another resourceGroup (template won't work for that as written).  The template can be authored for that but we may need changes in the CI engine to support if you want to support that.